### PR TITLE
Fix dynamic option type autocomplete

### PR DESCRIPTION
**Pull request by mbj**

### CODE REVIEW

**Comment by mbj** [comment]
@dkubb This is ready for review.

File: backend/app/views/spree/admin/products/_form.html.erb [hunk 1]
```diff
@@ -103,7 +103,19 @@
     <div data-hook="admin_product_form_option_types">
       <%= f.field_container :option_types do %>
         <%= f.label :option_type_ids, Spree.t(:option_types) %>
-        <%= f.hidden_field :option_type_ids, :value => @product.option_type_ids.join(',') %>
+        <%=
+          f.select(
+            :option_type_ids,
+            options_from_collection_for_select(
+              Spree::OptionType.all.order(:id),
```
Comment (dkubb) [inline backend/app/views/spree/admin/products/_form.html.erb]: Is the `.all` necessary?

Comment (dkubb) [inline backend/app/views/spree/admin/products/_form.html.erb]: Is the `.all` necessary?

Comment (mbj) [inline backend/app/views/spree/admin/products/_form.html.erb]: AFAIK yes to start a relation? I'll test it out.

Comment (dkubb) [inline backend/app/views/spree/admin/products/_form.html.erb]: I'm pretty sure it isn't necessary, `Spree::OptionType.order(:id)` should be enough to return a relation that can be used in this context.

**Comment by mbj** [comment]
@dkubb This is ready for the next pass.

**Pull request merged**